### ISSUE TITLE
GraphQL field ordering is _still_ not spec compliant

### ISSUE THREAD

**Issue by lutter** [graphprotocol/graph-node#3235]
When running a query like `{ _meta { block {  number }  } things { id }}` the response contains `things` first and `_meta` at the end. This is most likely because we pick fields apart [here](https://github.com/graphprotocol/graph-node/blob/master/graphql/src/execution/execution.rs#L258-L264) and then [put them together](https://github.com/graphprotocol/graph-node/blob/master/graphql/src/execution/execution.rs#L268-L286) without considering their ordering in the query.

**Comment by azf20** [comment]
Good catch - to confirm the response is `deterministic`, but not spec-compliant?

**Comment by azf20** [comment]
Added to the GraphQL API board @dotansimha
